com.reprezen.swagedit/src/com/reprezen/swagedit/model/NodeDeserializer.java

```diff
@@ -107,11 +105,10 @@ protected ValueNode deserializeValueNode(JsonParser p, DeserializationContext co
 
         Object v = context.readValue(p, Object.class);
 
-        ValueNode node = new ValueNode(model, parent, ptr, v);
+        ValueNode node = model.valueNode(parent, ptr, v);
```

Comment:
I really like the new `objectNode()`, `valueNode` and `arrayNode()` methods. Because they are public we can communicate their intention better by renaming them to something like `addObjectNode()` or `createObjectNode`